test/libsolidity/SemanticTest.cpp

```diff
@@ -212,15 +252,34 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 				);
 			}
 
-			bool outputMismatch = (output != test.call().expectations.rawBytes());
-			// Pre byzantium, it was not possible to return failure data, so we disregard
-			// output mismatch for those EVM versions.
-			if (test.call().expectations.failure && !m_transactionSuccessful && !m_evmVersion.supportsReturndata())
-				outputMismatch = false;
-			if (m_transactionSuccessful != !test.call().expectations.failure || outputMismatch)
-				success = false;
+			bytes expectationOutput;
+			if (test.call().expectations.builtin)
+			{
+				auto builtin = m_builtins[test.call().expectations.builtin->signature];
```

Comment:
Why do you use `test.call().expectation.builtin->signature` here, but `test.call().signature` above?

Reply:
Because this refers to the builtin on the "expectation" side.
Lets say we have something like this:
```
// smoke.test0 -> smoke.test1
```
`test.call().signature` refers to `smoke.test0` , where `test.call().expectation.builtin->signature` refers to `smoke.test1`.